sklearn/compose/_column_transformer.py

```diff
@@ -654,13 +654,22 @@ def _check_X(X):
 def _is_empty_column_selection(column):
     """
     Return True if the column selection is empty (empty list or all-False
-    boolean array).
-
+    boolean array or all-False list).
     """
+
+    # Coerce to a numpy array to make the below tests easier
+    column = np.array(column)
```

Comment:
This makes a copy of `column` which can be expensive when there are many columns are.

Reply:
good point.  I'm happy to close this PR and defer to yours: https://github.com/scikit-learn/scikit-learn/pull/17616

I'll probably make a PR (after you merge yours) to add my unit test `def test_issue_17586`, which I like and would really love to add to the tests.